src/Altinn.Apps/AppFrontend/react/shared/__tests__/utils/langugage.test.ts

```diff
@@ -101,4 +101,20 @@ describe('>>> src/Altinn.Apps/AppFrontend/react/shared/src/utils/language.ts', (
     textResource = mockTextResources.find((resource) => resource.id === 'mockId-1');
     expect(textResource.value).toEqual('Hello, cat!');
   });
+
+  it('+++ should replace multiple references to same value', () => {
+    mockTextResources = [
+      {
+        id: 'mockId',
+        value: 'This is a {0} apple. It will always be {0}. Yes, {0} is my favorite color.',
+        unparsedValue: 'This is a {0} apple. It will always be {0}. Yes, {0} is my favorite color.',
+        variables: [
+          { key: 'model.text.color', dataSource: 'dataModel.test' },
+        ],
+      },
+    ];
+    replaceTextResourceParams(mockTextResources, mockDataSources);
```

Comment:
This is only my opinion, but I find it a bit weird to have a modern text function that mutates the params that are passed.

Reply:
While I agree, creating deep copies of arrays comes with performance costs. Since the replace function is called on each change in form data this might be an issue. I think refactoring this is outside of the scope of this bug. But feel free to create an issue on this if this is something you think we should look into.